### PR TITLE
New Observe implementation based on experimental Californium observe impl.

### PR DESCRIPTION
**Pull request by sbernard31**
This is a new implementation of ObservationRegistry based on the californium [PR #16](https://github.com/eclipse/californium/pull/16).

This should allow us to deploy [Leshan in a Cluster](https://github.com/eclipse/leshan/wiki/Cluster).

### CODE REVIEW

File: leshan-server-cf/src/main/java/org/eclipse/leshan/server/californium/impl/LeshanServer.java [hunk 1]
```diff
@@ -131,7 +131,8 @@ public void registered(final Client client) {
 
         // default endpoint
         coapServer = new CoapServer();
-        nonSecureEndpoint = new CoapEndpoint(localAddress);
+        nonSecureEndpoint = new CoapEndpoint(localAddress, NetworkConfig.getStandard(), this.observationRegistry,
+                this.observationRegistry.getObserveRequestStore());
```
Comment (ingoschaal) [inline leshan-server-cf/src/main/java/org/eclipse/leshan/server/californium/impl/LeshanServer.java]: I'm trying to test this pull request, but in the specified californium PR #16 i cannot find the constructor with four arguments in class CoapEndpoint. I'm missing the CaliforniumObservationRegistry. Thereby getting an error here in LeshanServer.

**Comment by sbernard31** [comment]
I did some modification in Californium which are not yet integrated in this branch.
If you want to test this experimental code, you should use this commit https://github.com/eclipse/californium/pull/16/commits/cfeb71633855dc893585c80d004e7ff0e34c3785 waiting I integrate the last modification in Leshan.

**Comment by sbernard31** [comment]
@ingoschaal , you should be able to test it now.

**Comment by ingoschaal** [comment]
thank you there are no more errors

**Comment by sbernard31** [comment]
We decide to merge #93  and #89 in #113.

**Pull request closed without merging**